3/juliaanoonan/juliaanoonan3.py

```diff
@@ -0,0 +1,109 @@
+#Hartree-Fock, yeah baby
+
+import numpy as np
+import psi4.core
+import scipy.linalg as la
+
+class rhf(object):
+
+    def __init__(self, mol, mints):
+
+        self.mol = mol
+        self.mints = mints
+
+        # initialization 1: read in Vnuc, integrals
+
+        self.Vnuc = mol.nuclear_repulsion_energy()
+        self.natom = mol.natom()
+        self.charge = mol.molecular_charge() # 0 if neutral, -1 if single anion, etc
+        self.norb = mints.basisset().nbf()  # num (spatial) orbitals = num basis functions
+        self.nuclear_charges = [mol.Z(A) for A in range(self.natom)] # nuc charges by atom
+        self.nelec = sum(self.nuclear_charges) - self.charge #num e- = nuc charges - total charge
+        self.nocc = int(self.nelec/2)
+
+        self.S = np.array(mints.ao_overlap()) # overlap integrals
+        self.T = np.array(mints.ao_kinetic()) # kinetic energy integral
+        self.V = np.array(mints.ao_potential()) # electron-nuclear attraction integrals
+        self.g = np.array(mints.ao_eri()).transpose((0,2,1,3)) # electron-electron repulsion integrals, in physicist's notation
+
+        # initialization 2: form orthogonalizer X
+
+        self.X = mints.ao_overlap()
+        self.X.power(-0.5, 1.e-16)
+        self.X = np.array(self.X)
+
+        # initialization 3: set D = 0 as starting guess
+
+        self.D = np.zeros((self.norb,self.norb))
+
+        # form core Hamiltonian (is not altered)
+
+
+        self.h = self.T + self.V
+        np.save("g",self.g)
```

Comment:
In general this is a good idea. Saving the two-electron integral as a numpy binary file would allow you to quickly reload it (as opposed to recalculating it each time during debugging). However, with sto-3g on water I doubt this will make a noticeable difference.